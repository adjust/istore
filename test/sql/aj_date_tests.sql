SELECT aj_date_out('2012-01-32');
SELECT aj_date_out('2012-01-31');
SELECT aj_date_out('2012-02-29');
SELECT aj_date_out('2013-02-29');
SELECT aj_date_out('2013-00-28');
SELECT aj_date_out('2013-13-28');
SELECT aj_date_out('2013-113-28');
SELECT aj_date_out('2013-3-28');
SELECT aj_date_out('2011-03-28');
SELECT aj_date_out('2180-03-28');
SELECT aj_date_out('201a-11-28');
SELECT aj_date_out('2012-1a-28');
SELECT aj_date_out('2012-11-2a');
SELECT aj_date_out('aaaa-aa-aa');
SELECT aj_date_out('2030-03-28');
