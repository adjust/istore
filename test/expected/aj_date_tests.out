SELECT aj_date_out('2012-01-32');
ERROR:  day out of range: 32
LINE 1: SELECT aj_date_out('2012-01-32');
                           ^
SELECT aj_date_out('2012-01-31');
 aj_date_out 
-------------
 2012-01-31
(1 row)

SELECT aj_date_out('2012-02-29');
 aj_date_out 
-------------
 2012-02-29
(1 row)

SELECT aj_date_out('2013-02-29');
ERROR:  day out of range: 29
LINE 1: SELECT aj_date_out('2013-02-29');
                           ^
SELECT aj_date_out('2013-00-28');
ERROR:  month out of range: 0
LINE 1: SELECT aj_date_out('2013-00-28');
                           ^
SELECT aj_date_out('2013-13-28');
ERROR:  month out of range: 13
LINE 1: SELECT aj_date_out('2013-13-28');
                           ^
SELECT aj_date_out('2013-113-28');
ERROR:  could not parse date string: 2013-113-28
LINE 1: SELECT aj_date_out('2013-113-28');
                           ^
SELECT aj_date_out('2013-3-28');
ERROR:  could not parse date string: 2013-3-28
LINE 1: SELECT aj_date_out('2013-3-28');
                           ^
SELECT aj_date_out('2011-03-28');
ERROR:  year out of range: 2011
LINE 1: SELECT aj_date_out('2011-03-28');
                           ^
SELECT aj_date_out('2180-03-28');
ERROR:  year out of range: 2180
LINE 1: SELECT aj_date_out('2180-03-28');
                           ^
SELECT aj_date_out('201a-11-28');
ERROR:  year out of range: 201
LINE 1: SELECT aj_date_out('201a-11-28');
                           ^
SELECT aj_date_out('2012-1a-28');
ERROR:  day out of range: -28
LINE 1: SELECT aj_date_out('2012-1a-28');
                           ^
SELECT aj_date_out('2012-11-2a');
ERROR:  could not parse date string: 2012-11-2a
LINE 1: SELECT aj_date_out('2012-11-2a');
                           ^
SELECT aj_date_out('aaaa-aa-aa');
ERROR:  could not parse date string: aaaa-aa-aa
LINE 1: SELECT aj_date_out('aaaa-aa-aa');
                           ^
SELECT aj_date_out('2030-03-28');
 aj_date_out 
-------------
 2030-03-28
(1 row)

